internal/api/upload.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+	"os"
+)
+
+func (c *ApiClient) Upload(ctx Context, reportUploadType string, uploadDate string, email string, file *os.File) error {
```

Comment:
as I commented on `Download`, I think you could extract things to a `model.NewUpload` function, and use `switch` at the end

Reply:
done